### ISSUE TITLE
2.12.0 cannot reproduce an Apple hls stream:  Unexpected exception handling load completed

### ISSUE THREAD

**Issue by fpezzato** [google/ExoPlayer#8100]
### Issue description
ExoPlayer 2.12.0 can't reproduce:
`https://play.itunes.apple.com/WebObjects/MZPlay.woa/hls/playlist.m3u8?cc=GB&a=1528598221&id=71318377&l=en&aec=HD`

Note: The same stream is reproducible with ExoPlayer 2.11.8

### Reproduction steps
1. Checkout ExoPlayer 2.12.0 
2. Edit the demo app file 'media.exolist.json' adding this stream
``` 
      "uri": "https://play.itunes.apple.com/WebObjects/MZPlay.woa/hls/playlist.m3u8?cc=GB&a=1528598221&id=71318377&l=en&aec=HD"

```
3. Play such stream
4. Observe no video being played, error in logcat:
```
com.google.android.exoplayer2.demo E/LoadTask: Unexpected exception handling load completed
      java.lang.ArrayIndexOutOfBoundsException: length=13; index=-1
        at com.google.android.exoplayer2.trackselection.BaseTrackSelection.isBlacklisted(BaseTrackSelection.java:178)
        at com.google.android.exoplayer2.source.hls.HlsChunkSource$InitializationTrackSelection.updateSelectedTrack(HlsChunkSource.java:606)
        at com.google.android.exoplayer2.source.hls.HlsChunkSource.getNextChunk(HlsChunkSource.java:265)
        at com.google.android.exoplayer2.source.hls.HlsSampleStreamWrapper.continueLoading(HlsSampleStreamWrapper.java:662)
        at com.google.android.exoplayer2.source.hls.HlsSampleStreamWrapper.continuePreparing(HlsSampleStreamWrapper.java:246)
        at com.google.android.exoplayer2.source.hls.HlsMediaPeriod.buildAndPrepareSampleStreamWrappers(HlsMediaPeriod.java:521)
        at com.google.android.exoplayer2.source.hls.HlsMediaPeriod.prepare(HlsMediaPeriod.java:160)
        at com.google.android.exoplayer2.source.MaskingMediaPeriod.createPeriod(MaskingMediaPeriod.java:122)
        at com.google.android.exoplayer2.source.MaskingMediaSource.onChildSourceInfoRefreshed(MaskingMediaSource.java:209)
        at com.google.android.exoplayer2.source.MaskingMediaSource.onChildSourceInfoRefreshed(MaskingMediaSource.java:37)
        at com.google.android.exoplayer2.source.CompositeMediaSource.lambda$prepareChildSource$0$CompositeMediaSource(CompositeMediaSource.java:113)
        at com.google.android.exoplayer2.source.-$$Lambda$CompositeMediaSource$XIMxMigUUqAS2WvqSyL5xh9sCFo.onSourceInfoRefreshed(Unknown Source:4)
        at com.google.android.exoplayer2.source.BaseMediaSource.refreshSourceInfo(BaseMediaSource.java:84)
        at com.google.android.exoplayer2.source.hls.HlsMediaSource.onPrimaryPlaylistRefreshed(HlsMediaSource.java:565)
        at com.google.android.exoplayer2.source.hls.playlist.DefaultHlsPlaylistTracker.onPlaylistUpdated(DefaultHlsPlaylistTracker.java:375)
        at com.google.android.exoplayer2.source.hls.playlist.DefaultHlsPlaylistTracker.access$1000(DefaultHlsPlaylistTracker.java:46)
        at com.google.android.exoplayer2.source.hls.playlist.DefaultHlsPlaylistTracker$MediaPlaylistBundle.processLoadedPlaylist(DefaultHlsPlaylistTracker.java:654)
        at com.google.android.exoplayer2.source.hls.playlist.DefaultHlsPlaylistTracker$MediaPlaylistBundle.onLoadCompleted(DefaultHlsPlaylistTracker.java:547)
        at com.google.android.exoplayer2.source.hls.playlist.DefaultHlsPlaylistTracker$MediaPlaylistBundle.onLoadCompleted(DefaultHlsPlaylistTracker.java:462)
        at com.google.android.exoplayer2.upstream.Loader$LoadTask.handleMessage(Loader.java:481)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.os.HandlerThread.run(HandlerThread.java:67)
```

### Link to test content
Edit `media.exolist.json`, edit the `HLS` object as :
```
  {
    "name": "HLS",
    "samples": [
      {
        "name": "Apple test",
        "uri": "https://play.itunes.apple.com/WebObjects/MZPlay.woa/hls/playlist.m3u8?cc=GB&a=1528598221&id=71318377&l=en&aec=HD"
      },
  [.....]
```
### A full bug report captured from the device
[bugreport-sdk_gphone_x86_arm-RSR1.200819.001.A1-2020-10-21-11-36-43.zip](https://github.com/google/ExoPlayer/files/5415335/bugreport-sdk_gphone_x86_arm-RSR1.200819.001.A1-2020-10-21-11-36-43.zip)

### Version of ExoPlayer being used
2.12.0

### Device(s) and version(s) of Android being used
Repro rate: 100%
Affected Devices:

- Samsung Note 8  (OS: Android 9)
- Pixel XL  (OS: Android  10)
- Emulator  (OS: Android 11)

**Comment by fpezzato** [comment]
I tracked it down to `com.google.android.exoplayer2.source.hls.InitializationTrackSelection`
where  `selectedIndex = indexOf(group.getFormat(0));` is not being able to find a suitable format, but I don't have a deep enough knowledge of ExoPlayer to pursue the issue any further unfortunately, so I opened this bug.

I am more than happy to do any other test this issue might require, thank you!

**Comment by ojw28** [comment]
This is a duplicate of https://github.com/google/ExoPlayer/issues/8025. It will be fixed in 2.12.1, which is tentatively scheduled for later this week.
